web/src/components/AirgapUploadProgress.jsx

```diff
@@ -45,44 +46,52 @@ function AirgapUploadProgress(props) {
 
   let progressBar;
   let percentage;
+  let uploadComplete;
 
   if (total > 0 && sent > 0) {
+    uploadComplete = sent === total
     percentage = Math.floor((sent / total) * 100).toFixed() + "%";
     progressBar = (
-      <div className={`progressbar ${smallSize && "small"}`}>
-        <div className="progressbar-meter" style={{ width: `${(sent / total) * (smallSize ? 100 : 600)}px` }} />
+      <div className={`progressbar ${smallSize ? "small" : ""}`}>
+        <div className={`progressbar-meter ${uploadComplete ? "complete" : ""}`} style={{ width: `${(sent / total) * (smallSize ? 100 : 600)}px` }} />
       </div>
     );
   } else {
     percentage = "0%";
     progressBar = (
-      <div className={`progressbar ${smallSize && "small"}`}>
+      <div className={`progressbar ${smallSize ? "small" : ""}`}>
         <div className="progressbar-meter" style={{ width: "0px" }} />
       </div>
     );
   }
 
-  props.data?.startPolling(2000);
-
+  props.data?.startPolling(1000);
+  
   let statusMsg = getAirgapInstallStatus?.currentMessage;
   try {
     // Some of these messages will be JSON formatted progress reports.
-    const jsonMessage = JSON.parse(statusMsg);
+    let jsonMessage;
+    jsonMessage = JSON.parse(statusMsg);
     const type = get(jsonMessage, "type");
     if (type === "progressReport") {
-      statusMsg = jsonMessage.compatibilityMessage;
-      // TODO: handle image upload progress here
+      try {
+        const parsedMsg = JSON.parse(jsonMessage.compatibilityMessage);
```

Comment:
i think this is not needed anymore now that the api is updated. you can do `statusMsg = jsonMessage.compatibilityMessage;` and ignore on catch